web/modules/custom/joinup_eulogin/src/Plugin/Menu/EuLoginMenuLink.php

```diff
@@ -98,55 +22,24 @@ public function isEnabled(): bool {
    * {@inheritdoc}
    */
   public function getCacheContexts(): array {
+    $cache = parent::getCacheContexts();
     // This is different on every URL, including query arguments, and is hidden
     // for authenticated users.
-    return [
-      'url',
-      'url.query_args',
-      'user.roles:authenticated',
-    ];
+    $cache[] = 'user.roles:authenticated';
+
+    return $cache;
   }
```

Comment:
Cache metadata is not merged in this way. We should use: `\Drupal\Core\Cache\Cache::mergeContexts()`. The code can be simplified to:

```php
return Cache::mergeContexts(parent::getCacheContexts(), ['user.roles:authenticated']);
```